backend/src/test/java/org/uresti/pozarreal/service/impl/PaymentConceptsServiceImplTests.java

```diff
@@ -0,0 +1,149 @@
+        Assertions.assertThat(paymentConceptList.get(0).getId()).isEqualTo("id1");
+        Assertions.assertThat(paymentConceptList.get(0).getLabel()).isEqualTo("label1");
+        Assertions.assertThat(paymentConceptList.get(1).getId()).isEqualTo("id2");
+        Assertions.assertThat(paymentConceptList.get(1).getLabel()).isEqualTo("label2");
+    }
+
+    @Test
+    public void givenAnPaymentConcept_whenSave_thenPaymentConceptIsSaved() {
+        //Given:
+        PaymentConceptsRepository paymentConceptsRepository = Mockito.mock(PaymentConceptsRepository.class);
+
+        PaymentConceptsServiceImpl paymentConceptsService = new PaymentConceptsServiceImpl(paymentConceptsRepository);
+
+        ArgumentCaptor<org.uresti.pozarreal.model.PaymentConcept> argumentCaptor =
+                ArgumentCaptor.forClass(org.uresti.pozarreal.model.PaymentConcept.class);
+
+        PaymentConcept paymentConceptDto = PaymentConcept.builder()
+                .id("id1")
+                .label("label1")
+                .build();
+
+        org.uresti.pozarreal.model.PaymentConcept paymentConcept = org.uresti.pozarreal.model.PaymentConcept.builder()
+                .id("id2")
+                .label("label2")
+                .build();
+
+        Mockito.when(paymentConceptsRepository.save(argumentCaptor.capture())).thenReturn(paymentConcept);
+
+        //When:
+
+        PaymentConcept paymentConceptSaved = paymentConceptsService.save(paymentConceptDto);
+
+        //Then:
+        org.uresti.pozarreal.model.PaymentConcept parameter = argumentCaptor.getValue();
+
+        Assertions.assertThat(paymentConceptSaved).isNotNull();
+        Assertions.assertThat(parameter.getId()).isNotNull();
+        Assertions.assertThat(parameter.getId()).isNotEqualTo("id1");
+        Assertions.assertThat(parameter.getLabel()).isEqualTo("label1");
+        Assertions.assertThat(paymentConceptSaved.getId()).isEqualTo("id2");
+        Assertions.assertThat(paymentConceptSaved.getLabel()).isEqualTo("label2");
+
+        Mockito.verify(paymentConceptsRepository).save(parameter);
+        Mockito.verifyNoMoreInteractions(paymentConceptsRepository);
+    }
+
+    @Test
+    public void givenAnPaymentConcept_whenUpdate_thenPaymentConceptIsUpdated() {
+        //Given:
+        PaymentConceptsRepository paymentConceptsRepository = Mockito.mock(PaymentConceptsRepository.class);
+        PaymentConceptsServiceImpl paymentConceptsService = new PaymentConceptsServiceImpl(paymentConceptsRepository);
+
+        org.uresti.pozarreal.model.PaymentConcept paymentConcept = org.uresti.pozarreal.model.PaymentConcept.builder()
+                .id("id")
+                .label("label")
+                .build();
+
+        Mockito.when(paymentConceptsRepository.save(paymentConcept)).thenReturn(paymentConcept);
+
+        //When:
+        PaymentConcept paymentConceptSaved = paymentConceptsService.update(PaymentConceptMapper.entityToDto(paymentConcept));
```

Comment:
Don't use "PaymentConceptMapper.entityToDto(paymentConcept)" calling the update method that we are testing here... just pass the object built manually (without calling to entityToDto) otherwise if entityToDto has errors your test will be affected

Reply:
done